### PR TITLE
LCAM-909|Orchestrate Update Hardship flow

### PR DESCRIPTION
**Pull request by venkat980**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-909)

Orchestrate Update Hardship flow.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java [hunk 1]
```diff
@@ -39,34 +42,80 @@ public ApplicationDTO create(WorkflowRequest request) {
         // Need to refresh from DB as HardshipDetail ids may have changed
         HardshipReviewDTO newHardship = hardshipService.find(performHardshipResponse.getHardshipReviewId());
 
-        if (courtType == CourtType.MAGISTRATE) {
+        if (application.getCourtType() == CourtType.MAGISTRATE) {
```
[the remark this answers is not in the document]
Comment (venkat980) [reply maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java:45]: Added explicit check for CROWN Court.

File: maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java [hunk 2]
```diff
@@ -39,34 +42,80 @@ public ApplicationDTO create(WorkflowRequest request) {
...
             hardshipOverview.setMagCourtHardship(newHardship);
-            // TODO: Call assessments.determine_mags_rep_decision stored procedure
-            boolean isVariationRequired = contributionService.isVariationRequired(application);
-            if (isVariationRequired) {
-                application = contributionService.calculateContribution(request);
-            }
+            application = processMagCourtHardshipRules(request);
```
[the remark this answers is not in the document]
Comment (venkat980) [reply maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java:47]: This was existing logic. We need to check this further.

File: maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java [hunk 3]
```diff
@@ -39,34 +42,80 @@ public ApplicationDTO create(WorkflowRequest request) {
...
+
+        hardshipService.updateHardship(request);
+
+        if (request.getApplicationDTO().getCourtType() == CourtType.MAGISTRATE) {
+            AssessmentStatusDTO assessmentStatusDTO = request.getApplicationDTO().getAssessmentDTO().getFinancialAssessmentDTO()
+                    .getHardship().getMagCourtHardship().getAsessmentStatus();
+            if (assessmentStatusDTO != null && CurrentStatus.COMPLETE.getValue().equals(assessmentStatusDTO.getStatus())) {
+                request.setApplicationDTO(processMagCourtHardshipRules(request));
+            }
+        } else {
```
[the remark this answers is not in the document]
Comment (venkat980) [reply maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java:72]: Added explicit check for CROWN Court.

File: maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java [hunk 4]
```diff
@@ -39,34 +42,80 @@ public ApplicationDTO create(WorkflowRequest request) {
...
+    /**
+     * This method performs the logic from the following stored procedures:
+     * crown_court.check_crown_court_actions(p_application_object => p_application_object);
+     * application.update_cc_application(p_application_object => p_application_object);
+     */
+    private ApplicationDTO checkActionsAndUpdateApplication(WorkflowRequest request) {
+        request.setApplicationDTO(contributionService.calculateContribution(request));
+
+        // call application.pre_update_cc_application stored procedure
+        request.setApplicationDTO(maatCourtDataService.invokeStoredProcedure(request.getApplicationDTO(),
```
[the remark this answers is not in the document]
Comment (venkat980) [reply maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java:104]: Check crown court actions is done as part of the proceedingsService.updateApplication(request). Please refer to the CCP updateApplication service.

File: maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java [hunk 5]
```diff
@@ -39,34 +42,80 @@ public ApplicationDTO create(WorkflowRequest request) {
...
-
-            // TODO: Call application.handle_eform_result stored procedure
-
-            // TODO: Call crown_court.xx_process_activity_and_get_correspondence stored procedure
+            application = checkActionsAndUpdateApplication(request);
         }
 
         // Update assessment summary view - displayed on the application tab
-        AssessmentSummaryDTO hardshipSummary = assessmentSummaryService.getSummary(newHardship, courtType);
+        AssessmentSummaryDTO hardshipSummary = assessmentSummaryService.getSummary(newHardship, application.getCourtType());
```
[the remark this answers is not in the document]
Comment (venkat980) [reply maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java:54]: This was existing logic, we can check this further.

File: maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (venkat980) [reply maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/HardshipOrchestrationService.java]: As discussed updated the logic.

**Review by muthusubra — APPROVED** [review]

**Pull request merged**